src/commands/utils/detectors.ts

```diff
@@ -1,60 +1,66 @@
 import { Errorable, combine, failed, getErrorMessage } from "./errorable";
-import AksClusterTreeItem from '../../tree/aksClusterTreeItem';
-import * as fs from 'fs';
-import * as path from 'path';
-import { ARMResponse, CategoryDetectorARMResponse, SingleDetectorARMResponse, isCategoryDataset } from "../../webview-contract/webviewDefinitions/detector";
+import AksClusterTreeItem from "../../tree/aksClusterTreeItem";
+import * as fs from "fs";
+import * as path from "path";
+import {
+    ARMResponse,
+    CategoryDetectorARMResponse,
+    SingleDetectorARMResponse,
+    isCategoryDataset,
+} from "../../webview-contract/webviewDefinitions/detector";
 import { getResourceManagementClient } from "./clusters";
-const tmp = require('tmp');
-const meta = require('../../../package.json');
+import { dirSync } from "tmp";
+import meta from "../../../package.json";
 
 /**
  * Can be used to store the JSON responses for a collection of category detectors and all their child detectors.
  */
-export async function saveAllDetectorResponses(
-    target: AksClusterTreeItem,
-    categoryDetectorIds: string[]
-) {
-    const outputDirObj = tmp.dirSync();
-
-    function saveDetector(detector: ARMResponse<any>) {
-        const detectorFilePath = path.join(outputDirObj.name, `${detector.name}.json`);
-        // Anonymize the data.
-        detector.id = "/subscriptions/12345678-1234-1234-1234-1234567890ab/resourcegroups/test-rg/providers/Microsoft.ContainerService/managedClusters/test-cluster/detectors/" + detector.name;
-        fs.writeFileSync(detectorFilePath, JSON.stringify(detector, null, 2));
-    }
+export async function saveAllDetectorResponses(target: AksClusterTreeItem, categoryDetectorIds: string[]) {
```

Comment:
Just some minor refactoring to move the nested function out.

Note, this code is not run as part of any commands, and was put in for generating realistic test data from ARM calls.